museek/plugin/sanity_check_observation_plugin.py

```diff
@@ -220,3 +222,56 @@ def create_plots_of_scan_data(self, data: TimeOrderedData):
         plt.hist(data.elevation.squeeze.flatten(), bins=200)
         plt.xlabel('elevation')
         self.savefig(description='Elevation histogram of all dishes during scan.')
+
+
+    def check_closeness_to_sunrise_sunset(self, data: TimeOrderedData, report_writer: ReportWriter):
+
+        """
+        Check if observation was done within 30 min of sunset or sunrise.
+        :param data: the `TimeOrderedData` to check
+        :param report_writer: the `ReportWriter` object to handle the report
+        """
+
+        import ephem
+        from datetime import datetime, timedelta
+        import numpy as np
+
+        observer = ephem.Observer()
+        observer.lat = '-30.7130'  # Latitude of Meerkat
+        observer.lon = '21.4203'  # Longitude of Meerkat
+
+        # Set the observer's date
+        date_time = datetime.utcfromtimestamp(float(data.original_timestamps[0]))
+        observer.date = date_time
+
+        # Calculate sunset and sunrise times
+        sunset_time = observer.previous_setting(ephem.Sun())  # Sunset
+        sunrise_time = observer.next_rising(ephem.Sun())  # Sunrise
+        
+        # Convert times to South Africa local timezone
+        sunset_local_time = datetime.strptime(str(sunset_time), "%Y/%m/%d %H:%M:%S") + timedelta(hours=2)
```

Comment:
Do you need the `timedelta` ? It's added everywhere, so maybe it drops when you do the subtraction?

Reply:
I use the timedelta here to print the time in SAST later, even though it will be dropped during the subtraction. Or is  it better to print the time in UTC ?

Reply:
@mariogrs should we print time in UTC or SAST? `astrokat` uses UTC, so maybe we're more consistent with observation planning if we use UTC...

Reply:
does it matter? It's the time difference that matters

Reply:
I would opt for UTC because it can be the same even if we end up using this on other sites. SAST we'd need to change...

Reply:
Use UTC now